lib/component.js

```diff
@@ -148,6 +148,39 @@ class Component extends WebComponent {
     this.$panelRoot.router.navigate(...arguments);
   }
 
+  /**
+   * Helper function which will queue a function to be run once the component has been
+   * initialized and added to the DOM. If the component has already had its connectedCallback
+   * run, the function will run immediately.
+   *
+   * It can optionally return a function to be enqueued to be run just before the component is
+   * removed from the DOM. This occurs during the disconnectedCallback lifecycle.
+   * @param callback Function to be run after the component has been added to the DOM
```

Comment:
for generated docs, give type and param name:
```suggestion
   * @param {function} fn - callback to be run after the component has been added to the DOM. If this callback returns another function, the returned function will be run when the component disconnects from the DOM.
```
For public APIs it's also nice to give an `@example` usage; see the other methods above and below this.